projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts

```diff
@@ -196,6 +196,16 @@ export class IgxTooltipTargetDirective extends IgxToggleActionDirective implemen
 
         this._overlayDefaults.positionStrategy = new AutoPositionStrategy(positionSettings);
         this._overlayDefaults.closeOnOutsideClick = false;
+        this._overlayDefaults.closeOnEscape = true;
+
+        this.target.onClosing.subscribe((event) => {
```

Comment:
We should remove all subscriptions in `ngOnDestroy` to prevent any potential memory leaks.
Maybe something like this:
```typescript
private destroy$ = new Subject();

this.target.onClosing.pipe(takeUntil(this.destroy$)).subscribe(/* ... */);

public ngOnDestroy(): void {
   this.destroy$.next();
   this.destroy$.complete();
}	
```

Reply:
Done